evalml/data_checks/uniqueness_data_check.py

```diff
@@ -0,0 +1,105 @@
+                unique enough columns.
+
+        Example:
+            >>> import pandas as pd
+            >>> df = pd.DataFrame({
+            ...    'regression_unique_enough': [float(x) for x in range(100)],
+            ...    'regression_not_unique_enough': [float(1) for x in range(100)]
+            ... })
+            >>> uniqueness_check = UniquenessDataCheck(problem_type="regression", threshold=0.8)
+            >>> assert uniqueness_check.validate(df) == {"errors": [],\
+                                                         "warnings": [{"message": "Input columns (regression_not_unique_enough) for regression problem type are not unique enough.",\
+                                                                 "data_check_name": "UniquenessDataCheck",\
+                                                                 "level": "warning",\
+                                                                 "code": "NOT_UNIQUE_ENOUGH",\
+                                                                 "details": {"column": "regression_not_unique_enough", 'uniqueness_score': 0.0}}]}
+        """
+        messages = {
+            "warnings": [],
+            "errors": []
+        }
+
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        res = X.apply(UniquenessDataCheck.uniqueness_score)
+
+        if is_regression(self.problem_type):
+            not_unique_enough_cols = list(res.index[res < self.threshold])
+            messages["warnings"].extend([DataCheckWarning(message=warning_not_unique_enough.format(col_name,
+                                                                                                   self.problem_type),
+                                                          data_check_name=self.name,
+                                                          message_code=DataCheckMessageCode.NOT_UNIQUE_ENOUGH,
+                                                          details={"column": col_name, "uniqueness_score": res.loc[col_name]}).to_dict()
+                                         for col_name in not_unique_enough_cols])
+        elif is_multiclass(self.problem_type):
+            too_unique_cols = list(res.index[res > self.threshold])
+            messages["warnings"].extend([DataCheckWarning(message=warning_too_unique.format(col_name,
+                                                                                            self.problem_type),
+                                                          data_check_name=self.name,
+                                                          message_code=DataCheckMessageCode.TOO_UNIQUE,
+                                                          details={"column": col_name, "uniqueness_score": res.loc[col_name]}).to_dict()
+                                         for col_name in too_unique_cols])
+        return messages
+
+    @staticmethod
+    def uniqueness_score(col):
```

Comment:
Out of curiosity, is there a reference from where this uniqueness score calculation comes from?

Reply:
The Herfindahl–Hirschman Index!  The name is included and the source is here:
https://en.wikipedia.org/wiki/Herfindahl%E2%80%93Hirschman_Index